### PR TITLE
Implement new projections

### PR DESCRIPTION
**Pull request by ziotom78**
This PR is a WIP to implement new features for plotting maps. So far the following new projections have been implemented:

- `orthographic2` (stereo orthographic projection)
- `gnomonic`

I plan to add some more facilities to add scatter and line plots over maps.

### CODE REVIEW

File: src/projections.jl [hunk 1]
```diff
@@ -84,30 +84,59 @@ lat2colat, colat2lat
...
+    equiproj(lat, lon)
+
+Equirectangular projection. Given the latitude `lat` (in radians) and the
+longitude (in radians), return a tuple (Bool, Number, Number) where the
+first Boolean is a flag telling if the point falls within the projection (true)
+or not (false), and the two numbers are the x and y coordinates of the point
+on the projection plane (both are in the range [−1, 1]).
+"""
+function equiproj(lat, lon)
+    # The weird usage of mod2pi(x + π) - π allows to have angles in the range
```
Comment (giordano) [inline src/projections.jl]: Wouldn't `rem2pi(x, RoundNearest)` be equivalent?

File: src/projections.jl [hunk 2]
```diff
@@ -84,30 +84,59 @@ lat2colat, colat2lat
...
+    mollweideproj(lat, lon)
+
+Mollweide projection. Given the latitude `lat` (in radians) and the
+longitude (in radians), return a tuple (Bool, Number, Number) where the
+first Boolean is a flag telling if the point falls within the projection (true)
+or not (false), and the two numbers are the x and y coordinates of the point
+on the projection plane (both are in the range [−1, 1]).
+"""
+function mollweideproj(lat, lon)
+    (true, sin(lat), 2 / π * lon * cos(lat))
```
Comment (giordano) [inline src/projections.jl]: You could use `sincos(lat)` to compute the sine and the cosine at the same time (you already did so in other places)

File: src/projections.jl (hunk 1 above)
Comment (ziotom78) [reply src/projections.jl]: I did not realize there was this nice solution. Thanks!

File: src/projections.jl (hunk 2 above)
Comment (ziotom78) [reply src/projections.jl]: Sure, I have changed the code.

**Pull request merged**